src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -120,26 +125,19 @@ internal virtual void ReleaseNativeResource()
                     _overlapped = null;
                 }
 
-                // Ensure we're no longer set as the current AwaitableProvider (we may not have been to begin with).
+                // Ensure we're no longer set as the current ValueTaskSource (we may not have been to begin with).
                 // Only one operation at a time is eligible to use the preallocated overlapped
-                _strategy.CompareExchangeCurrentOverlappedOwner(null, this);
+                _strategy.TryToReuse(this);
```

Comment:
Where is .Reset being called on the ManualResetValueTaskSourceCore inside the ValueTaskSource?  I don't see that call anywhere, and without it, if you were actually reusing the instance I'd expect to see a bunch of exceptions.  Which makes me skeptical we're actually pooling this as expected.

Reply:
There is no, `Reset`, we just assign it to `default`:

https://github.com/dotnet/runtime/blob/4adf09623f379bfff6eb257d6e71935626f0d305/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs#L38-L42

Reply:
Please call Reset instead.  You can then also just set RunContinuationsAsynchronously in the VTS constructor rather than setting it here, as Reset will maintain that.  Using Reset will help to detect most invalid uses of the returned ValueTask.